demosplan/DemosPlanCoreBundle/Entity/Permission/AccessControlPermission.php

```diff
@@ -0,0 +1,141 @@
+use DemosEurope\DemosplanAddon\Contracts\Entities\OrgaInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\RoleInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\UuidEntityInterface;
+use demosplan\DemosPlanCoreBundle\Entity\CoreEntity;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * Stores the information about the current phase of a procedure.
+ * Currently there a two phases related to a procedure, therefore this Entity is related to the procedure twice.
+ *
+ * @ORM\Table(name="access_control_permissions")
+ *
+ * @ORM\Entity(repositoryClass="demosplan\DemosPlanCoreBundle\Repository\AccessControlPermissionRepository")
+ */
+class AccessControlPermission extends CoreEntity implements UuidEntityInterface, AccessControlPermissionInterface
+{
+    /**
+     * @ORM\Column(type="string", length=36, options={"fixed":true})
+     *
+     * @ORM\Id
+     *
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     *
+     * @ORM\CustomIdGenerator(class="\demosplan\DemosPlanCoreBundle\Doctrine\Generator\UuidV4Generator")
+     */
+    protected string $id;
+
+    /**
+     * @ORM\Column(name="permission", type="string", nullable=false)
+     */
+    protected string $permission = '';
+
+    /**
+     * @Gedmo\Timestampable(on="create")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private DateTime $creationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $modificationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $deletionDate;
+
+    /**
+     * @var OrgaInterface|null
+     *
+     * @ORM\ManyToOne(targetEntity="demosplan\DemosPlanCoreBundle\Entity\User\Orga")
+     *
+     * @ORM\JoinColumn(name="orga_id", referencedColumnName="_o_id", nullable=true, onDelete="SET NULL")
+     */
+    protected $organisation;
+
+    /**
+     * @var RoleInterface|null
+     *
+     * @ORM\ManyToOne(targetEntity="demosplan\DemosPlanCoreBundle\Entity\User\Role")
+     *
+     * @ORM\JoinColumn(referencedColumnName="_r_id", nullable=true, onDelete="SET NULL")
+     */
+    protected $role;
+
+    /**
+     * @var CustomerInterface|null
+     *
+     * @ORM\ManyToOne(targetEntity="demosplan\DemosPlanCoreBundle\Entity\User\Customer")
+     *
+     * @ORM\JoinColumn(referencedColumnName="_c_id", nullable=true, onDelete="SET NULL")
+     */
+    protected $customer;
+
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    public function getPermission(): string
+    {
+        return $this->permission;
+    }
+
+    public function setPermission(string $permission): void
+    {
+        $this->permission = $permission;
+    }
+
+    /**
+     * @param OrgaInterface $orga
+     */
+    public function setOrga($orga)
```

Comment:
strict types instead of @param hints would be nice.

Reply:
@MoritzMandler what do you mean by stric types?

Reply:
setOrga(?OrgaInterface $orga)

Reply:
but I think you need to change the property too:
protected ?OrgaInterface $organisation;

Reply:
done here https://github.com/demos-europe/demosplan-core/commit/219857ca07df9886f4701178c08933285e6b2b9f

Reply:
pls add the return type to all setters
`public function setRole(?RoleInterface $role): void`